yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/utils/DockerCloudLoadComparator.java

```diff
@@ -0,0 +1,23 @@
+package com.github.kostyasha.yad.utils;
+
+import com.github.kostyasha.yad.DockerCloud;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import static com.github.kostyasha.yad.utils.DockerFunctions.countCurrentDockerSlaves;
+
+/**
+ * Class used to sort a List of DockerCloud objects by load.  Least loaded is
+ * considered earlier in the sort order.
+ *
+ * @author Sam Gleske (@samrocketman on GitHub)
+ */
+public class DockerCloudLoadComparator implements Comparator<DockerCloud>, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public int compare(DockerCloud a1, DockerCloud a2) {
+        return Integer.compare(countCurrentDockerSlaves(a1), countCurrentDockerSlaves(a2));
```

Comment:
@samrocketman 1) your comparator doesn't handle == case that is written in javadoc contract 2) when it added resulted logic simplifies down to this.

Reply:
I'm still relatively new to Java so I'm not sure how to update it so that it does handle the `==` case.

Reply:
```
Finally, the implementor must ensure that compare(x, y)==0 implies that sgn(compare(x, z))==sgn(compare(y, z)) for all z.
It is generally the case, but not strictly required that (compare(x, y)==0) == (x.equals(y)). Generally speaking, any comparator that violates this condition should clearly indicate this fact. The recommended language is "Note: this comparator imposes orderings that are inconsistent with equals."
```
After i added comparison IDEA suggested to use Integer.compare that do absolutely the same. I guess when two arguments are the same they wouldn't be swapped in sort like in your case. Hope there is no infinite sorting right now...

Reply:
But i solved it here, could you review/try PR? If it ok then i can merge and release.

Reply:
Thanks for explaining to me.